apple/internal/bundle_package_type.bzl

```diff
@@ -0,0 +1,27 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Support for bundle package type used by Apple bundling rules."""
+
+# Bundle package type identifiers
+# These correspond to the CFBundlePackageType key in Info.plist files.
+bundle_package_type = struct(
+    application = "APPL",
+    bundle = "BNDL",
+    framework = "FMWK",
+    extension = "XPC!",
+    kernel_extension = "KEXT",
+    test_bundle = "BNDL",
```

Comment:
I wouldn't duplicate values here; a "test bundle" is a "(loadable) bundle" so listing it twice feels a bit redundant. Admittedly, "extension" vs. "xpc service" feel different enough that it's weird to pick just one and use it in both places... maybe "extension_or_xpc"?

I don't feel **super** strongly about this though.

Reply:
Fixed